src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs

```diff
@@ -208,13 +208,11 @@ private void ReplicateToDestination()
                         {
                             while (true)
                             {
-#if DEBUG
```

Comment:
So I think that we need to run the tests of v4.0 in Debug mode, as we have lots of Debug.Assert in the code base and that wasn't the case in v3.5 and before.